sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs

```diff
@@ -158,5 +149,56 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
 
             return builder;
         }
+
+        private static TracerProviderBuilder AddVendorInstrumentationIfPackageNotReferenced(this TracerProviderBuilder tracerProviderBuilder)
+        {
+            var vendorInstrumentationActions = new Dictionary<string, Action>
```

Comment:
IMO, Dictionary seems like overkill for this.
Not a blocker, but I wonder if this could be simpler.

Reply:
Basic implementation had lot of condition with if and switch. This simples it.